### PR TITLE
feature/ContolGTP

### PR DESCRIPTION
**Pull request by ThijsLuttikhuis**
### Summary
ControlGoToPos returns a vector2 now, idea is to publish a single command in stead of seperate ones with kick / rotate / velocity

Tested it on attacker, works pretty well!

### Checklist for the reviewer
This acts as a guide to help you and/or the reviewer to analyze the code. 

**functionality**  
- Is the code is properly tested?
- Is the code working as expected
- Is the code is understandable and easy to read
- Are proper abstractions made? (no duplicate code, nice use of functions/objects)

**style**  
- Variables/objects/classes have proper names
- Proper use of variables: nu unused vars, no magic numbers
- No possible memory leaks. Mutexes properly used when multithreading? possible pointer issues?
- Are smart pointers used?
- unused variables?
- magic numbers?
- well documented?
- No unused imports?
- No leftover code / unused comments
- Proper indentation
- Proper use of namespaces

### CODE REVIEW

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 1]
```diff
@@ -9,6 +11,8 @@ namespace rtt {
 namespace ai {
 namespace control {
 
+ControlGoToPos::ControlGoToPos() = default;
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: you can define that it is default immediately in the .h file

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 2]
```diff
@@ -25,118 +29,96 @@ void ControlGoToPos::clear(GoToType goToType) {
...
-    command.y_vel = static_cast<float>(delta.y);
-    publishRobotCommand(command);
+    return pid.controlPIR(error, robot->vel);
 }
 
-void ControlGoToPos::goToPosForce(RobotPtr robot, Vector2 &targetPos) {
-
+Vector2 ControlGoToPos::goToPosForce(RobotPtr robot, Vector2 &targetPos) {
+    Vector2 a;
+    return a;
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: you can replace this with `return {} `

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 3]
```diff
@@ -25,118 +29,96 @@ void ControlGoToPos::clear(GoToType goToType) {
...
+Vector2 ControlGoToPos::goToPosLuTh(RobotPtr robot, Vector2 &targetPos) {
 
-void ControlGoToPos::goToPosLowLevel(RobotPtr robot, Vector2 &targetPos) {
+    return gtpLuth.goToPos(std::move(robot), targetPos);
 
 }
-void ControlGoToPos::goToPosHighLevel(RobotPtr robot, Vector2 &targetPos) {
 
+Vector2 ControlGoToPos::goToPosLowLevel(RobotPtr robot, Vector2 &targetPos) {
+    Vector2 a;
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: idem

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 4]
```diff
@@ -25,118 +29,96 @@ void ControlGoToPos::clear(GoToType goToType) {
...
+    return a;
 }
-void ControlGoToPos::goToPosBezier(RobotPtr robot, Vector2 &targetPos) {
 
+Vector2 ControlGoToPos::goToPosHighLevel(RobotPtr robot, Vector2 &targetPos) {
+    Vector2 a;
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: idem

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 5]
```diff
@@ -25,118 +29,96 @@ void ControlGoToPos::clear(GoToType goToType) {
...
+    return a;
 }
 
-void ControlGoToPos::publishRobotCommand(roboteam_msgs::RobotCommand &command) {
-    ioManager.publishRobotCommand(command);
+Vector2 ControlGoToPos::goToPosBezier(RobotPtr robot, Vector2 &targetPos) {
+    Vector2 a;
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: Idem

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosBallControl.cpp [hunk 6]
```diff
@@ -3,10 +3,19 @@
...
+namespace control {
+
+
+Vector2 ControlGoToPosBallControl::goToPos(ControlGoToPosBallControl::RobotPtr robot,
         Vector2 &target) {
 
-    Command command;
-    command.id = robot->id;
-    return command;
+    Vector2 a;
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/controlGoToPos/ControlGoToPosBallControl.cpp]: idem

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp [hunk 7]
```diff
@@ -80,11 +79,9 @@ ControlGoToPosLuTh::Command ControlGoToPosLuTh::goToPos(RobotPtr robot, Vector2
         }
 
         if (! nicePath) {
-            command.use_angle = 0;
             Vector2 dir = (targetPos - robot->pos).normalize();
-            command.x_vel = static_cast<float>(dir.x*2.0f);
-            command.y_vel = static_cast<float>(dir.y*2.0f);
-            command.w = static_cast<float>(control::ControlUtils::calculateAngularVelocity(robot->angle, 0));
+            command.x = static_cast<float>(dir.x*2.0f);
```
Comment (mrlukasbos) [inline roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: change the name

File: roboteam_ai/src/control/ControlGoToPos.cpp (hunk 2 above)
Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/ControlGoToPos.cpp]: done

File: roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp (hunk 7 above)
Comment (ThijsLuttikhuis) [reply roboteam_ai/src/control/controlGoToPos/ControlGoToPosLuTh.cpp]: done

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 8]
```diff
@@ -1,3 +1,5 @@
+#include <utility>
```
Comment (Barisimre) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: hmm...

File: roboteam_ai/src/control/ControlGoToPos.cpp [hunk 9]
```diff
@@ -25,118 +29,92 @@ void ControlGoToPos::clear(GoToType goToType) {
...
-    publishRobotCommand(command);
+Vector2 ControlGoToPos::goToPosBallControl(RobotPtr robot, Vector2 &targetPos) {
+    return gtpBallControl.goToPos(std::move(robot), targetPos);
 }
 
-void ControlGoToPos::goToPosBasic(RobotPtr robot, Vector2 &targetPos) {
+Vector2 ControlGoToPos::goToPosBasic(RobotPtr robot, Vector2 &targetPos) {
 
-    if (! robot) return;
+    if (! robot) return {};
```
Comment (Barisimre) [inline roboteam_ai/src/control/ControlGoToPos.cpp]: maybe warn that this is sort of an error?

**Pull request merged**